lib/services/api/metric_rollups_service.rb

```diff
@@ -0,0 +1,28 @@
+module Api
+  class MetricRollupsService
+    REQUIRED_PARAMS = %w(resource_type capture_interval start_date).freeze
+
+    def self.query_metric_rollups(params)
+      validate_params(params)
+
+      params[:offset] ||= 0
+      params[:limit] ||= Settings.api.metrics_default_limit
+
+      start_date = params[:start_date].to_date
+      end_date = params[:end_date].try(:to_date)
+
+      MetricRollup.rollups_in_range(params[:resource_type], params[:resource_ids], params[:capture_interval], start_date, end_date)
+    end
+
+    def self.validate_params(params)
+      REQUIRED_PARAMS.each do |key|
```

Comment:
Since this service is larger than one method, WDYT about making it a stateful object? If you did that, you could for instance extract each validation into its own private method, with intention-revealing names, etc..